sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/JacksonJsonReader.java

```diff
@@ -0,0 +1,190 @@
+     * JSON.
+     * @param jsonString String that is being parsed by the {@code parser}. May be null if using another source for
+     * JSON.
+     * @param resetSupported Whether this {@link JsonReader} can be reset to the beginning of the JSON stream.
+     * @param jsonOptions The {@link JsonOptions} used to configure this {@link JacksonJsonReader}.
+     * @throws NullPointerException If {@code parser} is null.
+     */
+    JacksonJsonReader(JsonParser parser, byte[] jsonBytes, String jsonString, boolean resetSupported,
+        JsonOptions jsonOptions) {
+        this.parser = Objects.requireNonNull(parser,
+            "Cannot create a Jackson-based instance of com.azure.json.JsonReader with a null Jackson JsonParser.");
+        this.jsonBytes = jsonBytes;
+        this.jsonString = jsonString;
+        this.resetSupported = resetSupported;
+        this.jsonOptions = jsonOptions;
+        this.currentToken = mapToken(parser.currentToken(), null);
+    }
+
+    @Override
+    public JsonToken currentToken() {
+        return currentToken;
+    }
+
+    @Override
+    public JsonToken nextToken() throws IOException {
+        currentToken = mapToken(parser.nextToken(), currentToken);
+        return currentToken;
+    }
+
+    @Override
+    public void close() throws IOException {
+        parser.close();
+    }
+
+    @Override
+    public byte[] getBinary() throws IOException {
+        if (currentToken() == JsonToken.NULL) {
+            return null;
+        } else {
+            return parser.getBinaryValue();
+        }
+    }
+
+    @Override
+    public boolean getBoolean() throws IOException {
+        return parser.getBooleanValue();
+    }
+
+    @Override
+    public float getFloat() throws IOException {
+        return parser.getFloatValue();
+    }
+
+    @Override
+    public double getDouble() throws IOException {
+        return parser.getDoubleValue();
+    }
+
+    @Override
+    public int getInt() throws IOException {
+        return parser.getIntValue();
+    }
+
+    @Override
+    public long getLong() throws IOException {
+        return parser.getLongValue();
+    }
+
+    @Override
+    public String getString() throws IOException {
+        return parser.getValueAsString();
+    }
+
+    @Override
+    public String getFieldName() throws IOException {
+        return parser.currentName();
+    }
+
+    @Override
+    public void skipChildren() throws IOException {
+        parser.skipChildren();
+    }
+
+    @Override
+    public JsonReader bufferObject() throws IOException {
+        JsonToken currentToken = currentToken();
+        if (currentToken == JsonToken.START_OBJECT || currentToken == JsonToken.FIELD_NAME) {
+            String json = readRemainingFieldsAsJsonObject();
+            return AzureJsonUtils.createReader(json, jsonOptions);
+        } else {
+            throw new IllegalStateException("Cannot buffer a JSON object from a non-object, non-field name "
```

Comment:
should we log exceptions here (since now we can) ?

Reply:
Good catch... I'm not sure how this passed linting